<?xml version="1.0"?>
<launch>

    <remap from="/diagnostics" to="/as/diagnostics" />

    <remap from="/as/brake_cmd" to="/vehicle/brake_cmd" />
    <remap from="/as/brake_info_report" to="/vehicle/brake_info_report" />
    <remap from="/as/brake_report" to="/vehicle/brake_report" />
    <remap from="/as/disable" to="/vehicle/disable" />
    <remap from="/as/enable" to="/vehicle/enable" />
    <remap from="/as/gps/fix" to="/vehicle/gps_fix"/>
    <remap from="/as/steering_cmd" to="/vehicle/steering_cmd" />
    <remap from="/as/steering_report" to="/vehicle/steering_report" />
    <remap from="/as/surround_report" to="/vehicle/surround_report" />
    <remap from="/as/throttle_cmd" to="/vehicle/throttle_cmd" />
    <remap from="/as/throttle_report" to="/vehicle/throttle_report" />
    <remap from="/as/turn_signal_cmd" to="/vehicle/turn_signal_cmd" />
    <remap from="/as/gear_cmd" to="/vehicle/gear_cmd" />
    <remap from="/as/gear_report" to="/vehicle/gear_report" />
    <remap from="/as/wheel_speed_report" to="/vehicle/wheel_speed_report" />
    <remap from="/as/adas_input" to="/vehicle/adas_input" />

    <include file="$(find pacmod)/launch/pacmod.launch">
      <arg name="use_kvaser" value="true"/>
      <arg name="kvaser_hardware_id" value="27780" />
      <arg name="kvaser_circuit_id" value="0" />
      <arg name="pacmod_vehicle_type" value="LEXUS_RX_450H" />
    </include>

    <group ns="as">
        <node pkg="joy" type="joy_node" name="joystick">
            <param name="deadzone" value="0.01"/>
            <param name="autorepeat_rate" value="20.0"/>
        </node>

        <node pkg="joystick_vehicle_test" type="joystick_vehicle_test" name="joystick_vehicle_test"
            args="-f $(find joystick_vehicle_test)/joystick_vehicle_test.json" output="screen"/>

        <node pkg="speed_model" type="speed_model" name="speed_model"
            args="-f $(find speed_model)/core_speed_model/speed_model.json" output="screen"/>

        <node pkg="steering_model" type="steering_model" name="steering_model"
            args="-f $(find steering_model)/core_steering_model/steering_model.json" />

        <node pkg="ds_veh_controller" type="ds_veh_controller" name="veh_controller"
            args="-f $(find ds_veh_controller)/core_veh_controller/ds_veh_controller.json" />

        <node pkg="ds_veh_controller" type="ds_veh_interface" name="veh_interface"
            args="-f $(find ds_veh_controller)/core_veh_controller/ds_veh_interface.json" />
    </group>

</launch>



